src/nvim/os/unix_defs.h

```diff
@@ -2,7 +2,7 @@
 #define NVIM_OS_UNIX_DEFS_H
 
 // Windows doesn't have unistd.h, so we include it here to avoid numerous
-// instances of `#ifdef HAVE_UNISTD_H'.
+// instances of `#ifdef WIN32'.
 #include <unistd.h>
```

Comment:
As a middle ground, we could have granular headers like `os_unistd.h` whose contents are essentially:

```
#ifdef UNIX
# include <unistd.h>
#else
# include <io.h>
#endif
```

Then only `nvim/os_unistd.h` need be included in the various `.c` files. 

Would like to hear from @aktau @ZyX-I @elmart regarding this.